### ISSUE TITLE
Demon Forged Breastplate On struck effect doesn't function, may also be missing spell-coefficency. 

### ISSUE THREAD

**Issue by HasieEST** [chromiecraft/chromiecraft#3348]
### What client do you play on?

enUS

### Faction

- [x] Alliance
- [x] Horde

### Content Phase:

- [x] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [x] 60

### Current Behaviour

First off, Demon Forged Breastplate doesn't proc, when you are struck.
Secondly, I have a suspicions that it also doesn't scale with spell power currently.
Attaching a video under here:
https://youtu.be/kPiSK0vhYHM

### Expected Blizzlike Behaviour

Obviously it should proc, when you are struck for about 3% of the times, and it should scale for about 40% of your spell power.

### Source

https://www.youtube.com/watch?v=r02IrY0gNI0 For classic dungeon farming guide, which covers Demon Forged Breastplate Spellcoefficency.
https://wowpedia.fandom.com/wiki/Demon_Forged_Breastplate a wiki link, which outlines any patches that has been done to the item. Since there is no explicit statement, that the spell-coefficency is removed, it is acceptable to believe that it carried over all the way to Wrath client. 

### Steps to reproduce the problem

As you can see in the video, Test was done for about 1.3 to 1.4 Mil damage taken. Each hit was around 45 to 55 dmg which equals roughly 28 000 hits taken (give or take). 
Gear is outlined at the start and midway through the test, there is also a WA, which shows how much spell power entire gear had (500ish).
To have similar test environment, 1 pull entire zombie room in ZF, while wearing outlined gear in the video and proceed to have extensive test run.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/4ac565699fcb7b88fcc15db97ec9f14a0894f331

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by HasieEST** [comment]
Granted, low time spent testing, so there may be disparity...

**Comment by HasieEST** [comment]
Alright after talking with Tempered, did a bit of testing again.. Here is the link: https://youtu.be/QCHVeg0-HEk
TL;DR It felt really inconsistent, started with both on me, only shield proc'd, chest didnt. Took shield off, no leeching was happening. Reequipped chest only. Worked, added Shield Worked. Took Shield off, worked. There seems to be some inconsistency, but it is quite hard to replicate. What I can confirm though is that Life leeching doesn't scale on Chest (Shield fix isn't live on CC yet, but as far as I know that should be fixed now)

**Comment by Dejanos** [comment]
> Alright after talking with Temper, did a bit of testing again.. Here is the link: https://youtu.be/QCHVeg0-HEk TL;DR It felt really inconsistent, started with both on me, only shield proc'd, chest didnt. Took shield off, no leeching was happening. Reequipped chest only. Worked, added Shield Worked. Took Shield off, worked. There seems to be some inconsistency, but it is quite hard to replicate. What I can confirm though is that Life leeching doesn't scale on Chest (Shield fix isn't live on CC yet, but as far as I know that should be fixed now)

Testing something that theoretically, but not surely to proc 3 out of 100 times can be a very tedious and time consuming task, reminds me of my tests with the Blue Dragon trinket procs. My advice to you would be if you can find the item on Keira 3 database tool or check here on github, and check its code proc behavior for sure, you would save much more time than doing testing on something that takes more than thousands of tries to determine its behavior with a relative confidence.

**Comment by Gultask** [comment]
It's really bizarre indeed. Found the same, took shield off, no proc, re-equipped chest, procced, re-equipped shield, procced.

It also is 100% **not** scaling with spellpower. My SP in this test was 2370
![wired](https://user-images.githubusercontent.com/100873791/163407678-167121f3-33b3-4ff7-b1a1-c5a469b27a62.png)

**Comment by Gultask** [comment]
Found another source too mentioning the 0.4 SP coefficient.
I'll go ahead and confirm at least that part of the issue. I'll mention the lack of procs too just in case. But it's nice knowing that there's a *workaround*, albeit a weird one.

**Comment by UltraNix** [comment]
Fixed.
